scene/gui/button.cpp

```diff
@@ -101,7 +102,8 @@ void Button::_notification(int p_what) {
 			case DRAW_HOVER: {
 
 				style = get_stylebox("hover");
-				style->draw(ci, Rect2(Point2(0, 0), size));
+				if (!flat)
+					style->draw(ci, Rect2(Point2(0, 0), size));
```

Comment:
This is the description for **Flat** property.

> Set the _flat_ property of a Button. Flat buttons don't display decoration unless hovered or pressed.

and this changes made nothing is drawn even hovered or pressed state.
is it correct behavior?